product-backend/src/test/java/AuthServiceTest.java

```diff
@@ -0,0 +1,76 @@
+import org.junit.jupiter.api.Test;
+import org.kainos.ea.db.AuthDao;
+import org.kainos.ea.db.DatabaseConnector;
+import org.kainos.ea.exception.FailedToRegisterException;
+import org.kainos.ea.exception.InvalidEmailException;
+import org.kainos.ea.exception.InvalidPasswordException;
+import org.kainos.ea.exception.InvalidRoleIdException;
+import org.kainos.ea.model.User;
+import org.kainos.ea.service.AuthService;
+import org.kainos.ea.service.AuthValidator;
+
+import java.sql.SQLException;
+
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class AuthServiceTest {
```

Comment:
No happy path

When_User_IsValid_Expect_EmailReturned

 you should verify
-  authDao mock was called
-  email have been returned